packages/serverless-components/nextjs-component/src/component.ts

```diff
@@ -475,6 +507,58 @@ class NextjsComponent extends Component {
       }
     }
 
+    if (hasISRPages) {
+      const regenerationLambdaInput: LambdaInput = {
+        region: bucketRegion,
+        description: inputs.description
+          ? `${inputs.description} (API)`
+          : "Next.js Regeneration Lambda",
+        handler: inputs.handler || "index.handler",
+        code: join(nextConfigPath, REGENERATION_LAMBDA_CODE_DIR),
+        role: {
+          service: ["lambda.amazonaws.com"],
+          policy: {
+            ...defaultLambdaPolicy,
+            Statement: [
+              ...(defaultLambdaPolicy.Statement as Record<string, unknown>[]),
+              {
+                Effect: "Allow",
+                Resource: queue.arn,
+                Action: [
+                  "sqs:ReceiveMessage",
+                  "sqs:DeleteMessage",
+                  "sqs:GetQueueAttributes"
+                ]
+              }
+            ]
+          }
+        },
+        memory: readLambdaInputValue(
+          "memory",
+          "regenerationLambda",
+          512
+        ) as number,
+        timeout: readLambdaInputValue(
+          "timeout",
+          "regenerationLambda",
+          10
+        ) as number,
+        runtime: readLambdaInputValue(
+          "runtime",
+          "regenerationLambda",
+          "nodejs12.x"
+        ) as string,
+        name: bucketOutputs.name
```

Comment:
Should this be trying to extract the value from my config, instead of using the s3 bucket name?:
```
        name:
            regenerationLambda: should-be-used <-----
```